packages/dashboards/tests/integration/components/dashboard-dimension-selector-test.js

```diff
@@ -0,0 +1,91 @@
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import { A as arr } from '@ember/array';
+import { clickTrigger, selectChoose } from 'ember-power-select/test-support/helpers';
+import { settled } from '@ember/test-helpers';
+
+moduleForComponent('dashboard-dimension-selector', 'Integration | Component | dashboard dimension selector', {
+  integration: true
+});
+
+test('it renders with right options', async function(assert) {
+  assert.expect(3);
+
+  const dashboard = {
+    widgets: Promise.resolve([
+      {
+        requests: arr([
+          {
+            logicalTable: {
+              table: {
+                name: 'a',
+                timeGrain: 'day'
+              },
+              timeGrain: {
+                dimensions: [
+                  { name: 'dim1', longName: 'dim1', category: 'cat1' },
+                  { name: 'dim2', longName: 'dim2', category: 'cat2' }
+                ]
+              }
+            }
+          }
+        ])
+      },
+
+      {
+        requests: arr([
+          {
+            logicalTable: {
+              table: {
+                name: 'b',
+                timeGrain: 'day'
+              },
+              timeGrain: {
+                dimensions: [
+                  { name: 'dim3', longName: 'dim3', category: 'cat2' },
+                  { name: 'dim1', longName: 'dim1', category: 'cat1' }
+                ]
+              }
+            }
+          }
+        ])
+      }
+    ])
+  };
+
+  this.set('dashboard', dashboard);
+
+  this.set('changeme', function(selection) {
+    assert.deepEqual(
+      { dimension: 'dim1', longName: 'dim1', tables: ['a', 'b'] },
+      selection,
+      'Selection sends correct dimension object'
+    );
+  });
+
+  await this.render(hbs`{{dashboard-dimension-selector dashboard=dashboard onChange=changeme}}`);
+
+  await settled();
```

Comment:
is this needed?

Reply:
yeah it has to wait for the dropdown to pop up in the wormhole.